From bc71840ec3c75b8587eeed5134f4428c88195f8c Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 18 Jan 2026 12:08:06 -0600
Subject: [PATCH] fix: Icon install directory

`/usr/share/pixmaps` is a legacy icon directory and should not be used any more for icon installation. Icons instead should be installed to a size-specific subdirectory of `/usr/share/icons/hicolor`. In fact I'm pretty sure that `/usr/share/pixmaps` isn't indexed by most DEs so the icon installed there would never would have been used. The only reason this wasn't noticed before is because most DEs support and prefer SVG icons.

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 Makefile                           | 2 +-
 pkg/bin/generate_spec.sh           | 2 +-
 pkg/fedora-spec/lact-headless.spec | 2 +-
 pkg/fedora-spec/lact.spec          | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 34faca0d..5e544576 100644
--- a/Makefile
+++ b/Makefile
@@ -23,7 +23,7 @@ build-release-headless:
 install-resources:
 	install -Dm644 res/lactd.service $(DESTDIR)$(PREFIX)/lib/systemd/system/lactd.service
 	install -Dm644 res/io.github.ilya_zlobintsev.LACT.desktop $(DESTDIR)$(PREFIX)/share/applications/io.github.ilya_zlobintsev.LACT.desktop
-	install -Dm644 res/io.github.ilya_zlobintsev.LACT.png $(DESTDIR)$(PREFIX)/share/pixmaps/io.github.ilya_zlobintsev.LACT.png
+	install -Dm644 res/io.github.ilya_zlobintsev.LACT.png $(DESTDIR)$(PREFIX)/share/icons/hicolor/512x512/apps/io.github.ilya_zlobintsev.LACT.png
 	install -Dm644 res/io.github.ilya_zlobintsev.LACT.svg $(DESTDIR)$(PREFIX)/share/icons/hicolor/scalable/apps/io.github.ilya_zlobintsev.LACT.svg
 	install -Dm644 res/io.github.ilya_zlobintsev.LACT.metainfo.xml $(DESTDIR)$(PREFIX)/share/metainfo/io.github.ilya_zlobintsev.LACT.metainfo.xml
 
diff --git a/pkg/bin/generate_spec.sh b/pkg/bin/generate_spec.sh
index 0cbe7d25..8d7bfe99 100755
--- a/pkg/bin/generate_spec.sh
+++ b/pkg/bin/generate_spec.sh
@@ -94,8 +94,8 @@ make install PREFIX=/usr DESTDIR=%{buildroot}
 /usr/bin/lact
 /usr/lib/systemd/system/lactd.service
 /usr/share/applications/io.github.ilya_zlobintsev.LACT.desktop
+/usr/share/icons/hicolor/512x512/apps/io.github.ilya_zlobintsev.LACT.png
 /usr/share/icons/hicolor/scalable/apps/io.github.ilya_zlobintsev.LACT.svg
-/usr/share/pixmaps/io.github.ilya_zlobintsev.LACT.png
 /usr/share/metainfo/io.github.ilya_zlobintsev.LACT.metainfo.xml
 
 %changelog
diff --git a/pkg/fedora-spec/lact-headless.spec b/pkg/fedora-spec/lact-headless.spec
index 5e6d12b9..aeea0569 100644
--- a/pkg/fedora-spec/lact-headless.spec
+++ b/pkg/fedora-spec/lact-headless.spec
@@ -31,8 +31,8 @@ make install PREFIX=/usr DESTDIR=%{buildroot}
 /usr/bin/lact
 /usr/lib/systemd/system/lactd.service
 /usr/share/applications/io.github.ilya_zlobintsev.LACT.desktop
+/usr/share/icons/hicolor/512x512/apps/io.github.ilya_zlobintsev.LACT.png
 /usr/share/icons/hicolor/scalable/apps/io.github.ilya_zlobintsev.LACT.svg
-/usr/share/pixmaps/io.github.ilya_zlobintsev.LACT.png
 /usr/share/metainfo/io.github.ilya_zlobintsev.LACT.metainfo.xml
 
 %changelog
diff --git a/pkg/fedora-spec/lact.spec b/pkg/fedora-spec/lact.spec
index 72dd0034..7e90e304 100644
--- a/pkg/fedora-spec/lact.spec
+++ b/pkg/fedora-spec/lact.spec
@@ -31,8 +31,8 @@ make install PREFIX=/usr DESTDIR=%{buildroot}
 /usr/bin/lact
 /usr/lib/systemd/system/lactd.service
 /usr/share/applications/io.github.ilya_zlobintsev.LACT.desktop
+/usr/share/icons/hicolor/512x512/apps/io.github.ilya_zlobintsev.LACT.png
 /usr/share/icons/hicolor/scalable/apps/io.github.ilya_zlobintsev.LACT.svg
-/usr/share/pixmaps/io.github.ilya_zlobintsev.LACT.png
 /usr/share/metainfo/io.github.ilya_zlobintsev.LACT.metainfo.xml
 
 %changelog
